prover/src/fri/mod.rs

```diff
@@ -0,0 +1,180 @@
+use lambdaworks_math::field::element::FieldElement;
+use lambdaworks_math::field::fields::u64_prime_field::U64PrimeField;
+use lambdaworks_math::polynomial::Polynomial;
+
+const ORDER: u64 = 293;
+pub type F = U64PrimeField<ORDER>;
+pub type FE = FieldElement<F>;
+
+fn fold_polynomial(
+    poly: &Polynomial<FieldElement<F>>,
+    beta: &FieldElement<F>,
+) -> Polynomial<FieldElement<F>> {
+    let coef = poly.coefficients();
+    let even_coef: Vec<FieldElement<F>> = coef
+        .iter()
+        .enumerate()
+        .filter(|(pos, _)| pos % 2 == 0)
+        .map(|(_pos, v)| *v)
+        .collect();
+
+    // odd coeficients of poly are multiplied by beta
+    let odd_coef_mul_beta: Vec<FieldElement<F>> = coef
+        .iter()
+        .enumerate()
+        .filter(|(pos, _)| pos % 2 == 1)
+        .map(|(_pos, v)| (*v) * beta)
+        .collect();
+
+    let (even_poly, odd_poly) = Polynomial::pad_with_zero_coefficients(
+        &Polynomial::new(&even_coef),
+        &Polynomial::new(&odd_coef_mul_beta),
+    );
+    even_poly + odd_poly
+}
+
+fn next_domain(input: &[FE]) -> Vec<FE> {
+    let length = input.len() / 2;
+    let mut ret = Vec::with_capacity(length);
+    for v in input.iter().take(length) {
+        ret.push(v * v)
+    }
+    ret
+}
+
+fn next_layer(poly: &Polynomial<FieldElement<F>>, domain: &[FE]) -> Vec<FE> {
+    let length = domain.len() / 2;
+    let mut ret = Vec::with_capacity(length);
+    for v in domain.iter() {
+        ret.push(poly.evaluate(v));
+    }
+    ret
+}
+
+/// Returns:
+/// * new polynomoial folded with FRI protocol
+/// * new domain
+/// * evaluations of the polynomial
+pub fn next_fri_layer(
+    poly: &Polynomial<FieldElement<F>>,
+    domain: &[FE],
+    beta: &FieldElement<F>,
```

Comment:
I see this normally named as "alpha", did you take this nomenclature from somewhere?

Reply:
Stark 101 uses beta for the randomness involved in FRI; alpha is used as the randomness to combine the polynomial constraints. Maybe we should make more comments inside the code so that it is clear.

Reply:
I saw beta both here https://medium.com/starkware/low-degree-testing-f7614f5172db and here https://github.com/facebook/winterfell/blob/cd76df242fcec8bab45efcdabf366fb0d8c76fbd/fri/src/folding/mod.rs#L57

Reply:
I believe it's better to use beta in this case, so it is easier to follow from stark 101. We will make sure that there is a comment explaining each variable and we can then add a note on other naming conventions.

Reply:
we can use any of the two but add comments and links.